oracle_fdw.c

```diff
@@ -771,11 +836,6 @@ oracleGetForeignRelSize(PlannerInfo *root, RelOptInfo *baserel, Oid foreigntable
 								&(fdwState->local_conds),
 								&(fdwState->remote_conds)
 							);
-
-	/* release Oracle session (will be cached) */
-	pfree(fdwState->session);
-	fdwState->session = NULL;
-
```

Comment:
Why did you remove that?

Reply:
It is not removed it is moved to oracleGetForeignPaths() after the append of limit clause to be able to read the Oracle version information with the call to min_oracle_version() If the session information is freed before we have a problem.

Reply:
Ah, ok, that makes sense. I did not see that.